### DISCUSSION TITLE
Feature request: make get-qe results that include terms over reals use decimals

### DISCUSSION THREAD

**Discussion by daniel-larraz** [cvc5/cvc5#7859] [Ideas]
Hello. Consider this dummy example:
```
(set-logic LRA)  
(declare-fun x () Real)
(get-qe
 (exists ((Y Bool)) (>= (* (- 1.0) x) (- 100.0))
))
```
Currently CVC5 returns:
`(>= (* (- 1) x) (- 100))`
But I would like CVC5 to return:
`(>= (* (- 1.0) x) (- 100.0))`
Or a term that uses rational numbers, but not numerals.
This would make the parsing and type inference of the expressions occurring in the resulting term easier for external tools that read the result, in particular, when other theories like integers are involved.
Thanks!

**Comment by ajreynol** [comment]
Indeed, this is an issue due to the way arithmetic subtyping works internally.  We are working towards eliminating arithmetic subtyping, although this is a long term project.

It may also be possible to patch the results of QE so they are strictly typed.

**Comment by daniel-larraz** [comment]
> It may also be possible to patch the results of QE so they are strictly typed.

This would be very helpful. We are working on making some functionality provided by Kind 2 compatible with Z3 and CVC5, and this would simplify our lives. Should we create an issue from this discussion?

**Comment by ajreynol** [comment]
Yes, go ahead and create an issue.
